pkg/dcmtime/datetime_test.go

```diff
@@ -0,0 +1,1344 @@
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.45678",
+			ExpectedString: "1010-02-03 04:05:06.45678",
+
+			Precision: PrecisionMS5,
+			NoOffset:  true,
+		},
+		// -2 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.4567-0102",
+			ExpectedString: "1010-02-03 04:05:06.4567 -01:02",
+			Precision:      PrecisionMS4,
+			NoOffset:       false,
+		},
+		// -2 Precision, no offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.4567",
+			ExpectedString: "1010-02-03 04:05:06.4567",
+			Precision:      PrecisionMS4,
+			NoOffset:       true,
+		},
+		// -3 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.456-0102",
+			ExpectedString: "1010-02-03 04:05:06.456 -01:02",
+			Precision:      PrecisionMS3,
+			NoOffset:       false,
+		},
+		// -3 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.456",
+			ExpectedString: "1010-02-03 04:05:06.456",
+			Precision:      PrecisionMS3,
+			NoOffset:       true,
+		},
+		// -4 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.45-0102",
+			ExpectedString: "1010-02-03 04:05:06.45 -01:02",
+			Precision:      PrecisionMS2,
+			NoOffset:       false,
+		},
+		// -4 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.45",
+			ExpectedString: "1010-02-03 04:05:06.45",
+			Precision:      PrecisionMS2,
+			NoOffset:       true,
+		},
+		// -5 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.4-0102",
+			ExpectedString: "1010-02-03 04:05:06.4 -01:02",
+			Precision:      PrecisionMS1,
+			NoOffset:       false,
+		},
+		// -5 Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
+			ExpectedDCM:    "10100203040506.4",
+			ExpectedString: "1010-02-03 04:05:06.4",
+			Precision:      PrecisionMS1,
+			NoOffset:       true,
+		},
+		// -6 Precision, with offset
```

Comment:
So while that is true, the IDE I use (Goland) makes test navigation MUCH easier to navigate if there are concrete tests that are being run.

When I run table-tests in Goland like this, I can double-click the failed test and it will navigate to the test case for me. It also lets you run a specific case out of an array like this if you want to work on one particular case.

That being said, I realize that structuring code tailored to an IDE is a contentious practice in some circle, I personally feel it makes me more productive.

I also just like having all the test cases explicitly laid out like this, it makes the tests easier to reason about for new maintainers down the line: If a test fails you don't have to reason about the test generation logic before you can even troubleshoot the case that is failing itself.

All that being said, I am more than happy to turn this into a generative set of tests.

Reply:
Eh, I don't feel too strongly on this one. I use GoLand too, and it's a very nice IDE. Feel free to leave it as is if you prefer!

Reply:
Makes sense to me!